### PR TITLE
OPENEUROPA-1385: Update oe_authentication and oe_authorisation

### PR DESCRIPTION
**Pull request by voidtek**
# OPENEUROPA-1385

## Description
Update oe_authentication and oe_authorisation

## Changelog

- [x] Update to the last version of both components
- [x] Update drone and docker files, and fixture files
- [x] Update behat tests
- [x] Update documentation

### CODE REVIEW

**Comment by drupol** [comment]
Good to read: https://medium.com/@danielfeelfine/commit-verbs-101-why-i-like-to-use-this-and-why-you-should-also-like-it-d3ed2689ef70

File: docker-compose.yml [hunk 1]
```diff
@@ -69,15 +81,3 @@ services:
...
-
-#volumes:
-#  nfsmount:
-#    driver: local
-#    driver_opts:
-#      type: nfs
-#      o: addr=host.docker.internal,rw,nolock,hard,nointr,nfsvers=3
-#      device: ":${PWD}/"
-
-#### End Mac users.
```
Comment (ademarco) [inline docker-compose.yml]: Shouldn't we keep this part for Mac users?

File: features/authentication.feature [hunk 2]
```diff
@@ -1,45 +1,24 @@
...
-      | Landing page Page principale has been updated. |
-    And I should see "Página principal" in the "page header"
-    And I should see "Esta es la página principal del sitio" in the "page header"
+    # Redirected to the Ecas mockup server.
+    And I press the "Log me out" button
 
-    When I am on "the logout page"
-    Then I should not see the link "My account"
+    # Redirected back to Drupal.
+    Then I should be on "/en"
```
Comment (ademarco) [inline features/authentication.feature]: We should use the transformation context to turn `Then I should be on "/en"` into `Then I should be on "the front page"` so it's more business friendly.

File: features/authorisation.feature [hunk 3]
```diff
@@ -1,53 +1,81 @@
-@api @site-inea
+# Tested only on info site, the syncope container is shared between
+# containers and others behat tests can interfere.
+@api @site-info @javascript
```
Comment (ademarco) [inline features/authorisation.feature]: Do we need to run all scenarios using `@javascript`?

File: features/editorial/landing-page.feature [hunk 4]
```diff
@@ -1,31 +1,39 @@
-@api @site-inea @site-energy @site-info
+@javascript @site-inea @site-energy @site-info
```
Comment (ademarco) [inline features/editorial/landing-page.feature]: Same here, is `@javascript` needed? And shouldn't we keep `@api`?

File: features/global_authorisation.feature [hunk 5]
```diff
@@ -1,22 +1,30 @@
+# @todo: Change the @site-info tag to @sites-all tag.
 @api @site-info
 Feature: Global authorisation
   In order to have global roles
   As a role manager
   I want to assign a global role to a user and have it propagated on all sites for that user
 
   @javascript
-  Scenario Outline: Global roles should be inherited by users on all sites
```
Comment (ademarco) [inline features/global_authorisation.feature]: Is `@javascript` needed?

File: features/rdf-content/content-broker.feature [hunk 6]
```diff
@@ -64,6 +66,7 @@ Feature: Content broker
 
   @site-info
   Scenario: The INFO only consumes announcements from the INEA site
+    # @todo: fix the identification.
```
Comment (ademarco) [inline features/rdf-content/content-broker.feature]: We should open a follow up for all these todos, do we already have an idea on how to fix it?

File: features/authentication.feature [hunk 7]
```diff
@@ -1,45 +1,24 @@
...
+  Scenario: As an editor I want to login via EU Login and create and translate a page
     When I am on "the EU Login page"
 
-    # Redirecting to the authentication mock service.
-    And I fill in "User" with "John Smith"
-    And I fill in "Nickname" with "The editor"
-    And I fill in "Password" with "john"
-    And I press the "Submit" button
+    # Redirected to the mock server.
+    And I click "European Commission" if exists
```
Comment (ademarco) [inline features/authentication.feature]: I'm not sure about a step that does something only if a certain condition is satisfied, why would we need this exactly?

File: src/Behat/Services/Manager/CasAuthenticationManager.php [hunk 8]
```diff
@@ -25,10 +25,14 @@ public function logIn(\stdClass $user): void {
     $this->getSession()->visit($this->locatePath('/en/caslogin'));
     $element = $this->getSession()->getPage();
 
-    $element->fillField("User", $user->name);
-    $element->fillField("Nickname", $user->name);
-    $element->fillField("Password", $user->pass);
-    $submit = $element->findButton("Submit");
+    if ($link = $element->findLink('European Commission')) {
+      $link->click();
+    }
```
Comment (ademarco) [inline src/Behat/Services/Manager/CasAuthenticationManager.php]: As above: why do we need this?

File: docker-compose.yml (hunk 1 above)
Comment (voidtek) [reply docker-compose.yml]: We can put again this part on the file. But we should organise a meeting to talk and organize this information.

File: features/authorisation.feature (hunk 3 above)
Comment (voidtek) [reply features/authorisation.feature]: @javascript is mandatory to log in. Authentication uses SSL and this tag does have the issue with the curl authorisation access.

File: features/editorial/landing-page.feature (hunk 4 above)
Comment (voidtek) [reply features/editorial/landing-page.feature]: We should keep `@javascript`

File: features/global_authorisation.feature (hunk 5 above)
Comment (voidtek) [reply features/global_authorisation.feature]: We should keep `@javascript`

File: features/rdf-content/content-broker.feature (hunk 6 above)
Comment (voidtek) [reply features/rdf-content/content-broker.feature]: I am preparing all followup tickets for behat test and the authentication steps. These followup tickets will be open on `oe_authentication` and behat trait.

File: features/authentication.feature (hunk 7 above)
Comment (voidtek) [reply features/authentication.feature]: This link should be clicked once during the test. So if you run only this scenario, you should click on the link. If you are running all behat tests, the link doesn't appear. This custom step was created to provide the possibility to run this scenario alone or with other scenarios.
I will include the reflexion of this step on the followup tickets.

File: src/Behat/Services/Manager/CasAuthenticationManager.php (hunk 8 above)
Comment (voidtek) [reply src/Behat/Services/Manager/CasAuthenticationManager.php]: Same response as above.

**Pull request merged**